cmake/modules/hunter_pack_git_submodule.cmake

```diff
@@ -193,7 +194,17 @@ function(hunter_pack_git_submodule)
   set(archive "${archives_directory}/${PACKAGE_NAME}.tar")
   hunter_status_debug("Creating archive '${archive}'")
 
-  set(cmd "${GIT_EXECUTABLE}" archive HEAD -o "${archive}")
+  # check if whole submodule or just a subfolder is to be archived
+  string(COMPARE EQUAL "${x_SUBMODULE_SOURCE_SUBDIR}" "" is_empty)
+  if(is_empty)
+    hunter_status_debug("No SUBMODULE_SOURCE_SUBDIR specified, archive whole submodule")
+    set(_submodule_source_flag)
```

Comment:
We are inside function which has separate scope. There is no need to use names started with underscore. `source_flags` will work just fine.